cpp/tests/utilities/column_utilities.cu

```diff
@@ -34,26 +39,97 @@ void expect_column_properties_equal(cudf::column_view lhs, cudf::column_view rhs
   EXPECT_EQ(lhs.size(), rhs.size());
   EXPECT_EQ(lhs.null_count(), rhs.null_count());
   if (lhs.size() > 0) {
-      EXPECT_EQ(lhs.nullable(), rhs.nullable());
+     EXPECT_EQ(lhs.nullable(), rhs.nullable());
   }
   EXPECT_EQ(lhs.has_nulls(), rhs.has_nulls());
   EXPECT_EQ(lhs.num_children(), rhs.num_children());
 }
 
-// Verify elementwise equality
-void expect_columns_equal(cudf::column_view lhs, cudf::column_view rhs) {
+class corresponding_rows_unequal {
+public:
+  corresponding_rows_unequal(table_device_view d_lhs, table_device_view d_rhs): comp(d_lhs, d_rhs) {
+  }
+  
+  cudf::experimental::row_equality_comparator<true> comp;
+    
+  __device__ bool operator()(size_type index) {
+    return !comp(index, index);
+  }
+};
+
+void expect_columns_equal(cudf::column_view lhs, cudf::column_view rhs, bool print_all_differences) {
   expect_column_properties_equal(lhs, rhs);
 
   auto d_lhs = cudf::table_device_view::create(table_view{{lhs}});
   auto d_rhs = cudf::table_device_view::create(table_view{{rhs}});
 
-  EXPECT_TRUE(
-      thrust::equal(thrust::device, thrust::make_counting_iterator(0),
-                    thrust::make_counting_iterator(lhs.size()),
-                    thrust::make_counting_iterator(0),
-                    cudf::experimental::row_equality_comparator<true>{*d_lhs, *d_rhs}));
+  thrust::device_vector<int> differences;
+  differences.reserve(lhs.size());
```

Comment:
I don't believe this is right. The `copy_if` will not increase the size of `differences`, so while the memory is allocated, `differences.size()` will always be zero.

Reply:
You are correct.  I could change it to resize().  I didn't worry about it because differences is only internal and I never access differences.size(), I use the end pointer returned from copy_if.

Reply:
I'd just do:
```
 thrust::device_vector<int> differences(lhs.size());
```
and call it a day.

Reply:
Actually, resize() would just populate it all with zeros and size would still be wrong.  I would need to compute the distance after the fact and then resize to that length.  Not sure what the purpose would be.

Reply:
Sure, I'd just call `resize` after the `copy_if`. I just don't like leaving data structures in a strange state for the next person who comes along and modifies the code.